### PR TITLE
Add SROS Support for Nokia IXR-X1, IXR-Xs, IXR-X3

### PR DESCRIPTION
**Pull request by sdktr**
Suggested hardware combo for the IXR-X1, IXR-Xs and IXR-X3 platforms

https://documentation.nokia.com/sr/22-10/books/vsim-installation-setup/appendixes.html#ai89jtwgeh

todo: needs  check from maintainers

### CODE REVIEW

**Comment by hellt** [comment]
Thanks @sdktr 
Can you post here the topoly file with a single sros image using this new variant and showing 'show version' from it?

**Comment by sdktr** [comment]
re-build the lab, first test with literal/custom definition. WIll test with my patch later. 

Does anyone know whether the 'show chassis' should output 7250 IXR-x for this hardware?

```yaml
topology:
  nodes:
    X:
      mgmt-ipv4: 192.168.8.101
      kind: vr-sros
      type: 'cpu=2 ram=4 slot=A chassis=ixr-x card=cpm-ixr-x/imm32-qsfp28+4-qsfpdd'
      image: vrnetlab/vr-sros:23.10.R1
      runtime: docker
      license: ../../sros/license-sros23.3.R3-202310.txt
```

```
A:admin@X# show chassis 
===============================================================================
System Information
===============================================================================
  Name                              : bng1
  Type                              : 7250 IXR-x
  Chassis Topology                  : Standalone
  Location                          : (Not Specified)
  Coordinates                       : (Not Specified)
  CLLI code                         :           
  Number of slots                   : 2
  Oper number of slots              : 2
  Num of faceplate ports/connectors : 36
  Num of physical ports             : 0
  Critical LED state                : Off
  Major LED state                   : Off
  Minor LED state                   : Off
  Over Temperature state            : OK
  Base MAC address                  : 0c:00:0b:4c:a9:00

===============================================================================
Chassis Summary
===============================================================================
Chassis   Role                Status
-------------------------------------------------------------------------------
1         Standalone          up
===============================================================================

A:admin@X# show version
TiMOS-C-23.10.R1 cpm/x86_64 Nokia 7250 IXR Copyright (c) 2000-2023 Nokia.
All rights reserved. All use subject to applicable license agreements.
Built on Thu Oct 26 20:12:19 UTC 2023 by builder in /builds/2310B/R1/panos/main/sros

[/]
```

**Comment by jcpvdm** [comment]
> Does anyone know whether the 'show chassis' should output 7250 IXR-x for this hardware?
@sdktr 
I think so, I have seen same output from real IXR-X1.

**Comment by hellt** [comment]
looks good to me then @sdktr 
if the PR is ready you can remove the draft status and I can merge it

File: sros/docker/launch.py [hunk 1]
```diff
@@ -151,17 +151,65 @@ def line_card_config(
...
+            "min_ram": 4,
+            "timos_line": "slot=A chassis=ixr-x card=cpm-ixr-x",
+        },
+        # line card (IOM/XCM)
+        "lcs": [
+            {
+                "min_ram": 4,
+                "timos_line": "chassis=ixr-x slot=1 card=imm36-qsfpdd",
+                "card_config": """
+            """,
```
Comment (hellt) [inline sros/docker/launch.py:190]: @sdktr you don't need to configure the cards for x?

Comment (sdktr) [reply sros/docker/launch.py:190]: thanks for pointing out the probable cause for my deployment failing. Hold on

**Comment by hellt** [comment]
Hi @sdktr 
is this ready now?

**Comment by sdktr** [comment]
Yes! The styling code checks failing are not related to this pr

**Comment by hellt** [comment]
thanks
yes, don't look at the deepsource, it is useless here at this moment

**Pull request merged**